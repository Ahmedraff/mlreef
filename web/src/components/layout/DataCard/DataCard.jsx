import React from 'react';
import {
  string, arrayOf, bool, shape,
} from 'prop-types';
import './DataCard.scss';

const DataCard = ({ title, linesOfContent }) => (
  <div className="data-card">
    <div className="title">
      <p><b>{title}</b></p>
    </div>
    <div>
      {linesOfContent?.map((line) => {
        const lineContent = line?.text?.startsWith('*')
          ? <b>{line.text.replace('*', '')}</b>
          : line.text;
        return line?.isLink
          ? (
            <a key={line?.text} target="_blank" rel="noopener noreferrer" href={line.href}>
              <b>{lineContent}</b>
            </a>
          )
          : <p key={line?.text} className="line">{lineContent}</p>;
      })}
    </div>
  </div>
);

DataCard.propTypes = {
  title: string.isRequired,
  linesOfContent: arrayOf(
    shape({
      text: string.isRequired,
      isLink: bool,
    }),
  ).isRequired,
};

export default DataCard;
