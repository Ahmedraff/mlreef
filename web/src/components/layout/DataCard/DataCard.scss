@import '../../../styles/breakpoints';

$Algorithm: #d2519d;
$visualizations: #735da8;

.data-card {
  border: 1px solid var(--almostWhite);
  margin: 0% 0% 1% 0.5%;
  border-radius: 0.5vw;
  height: 100%;

  @media (max-width: $bp-md) {
    height: auto;
  }

  &-rows {
    padding: 1rem;
  }

  > .model {
    background-color: $Algorithm !important;
  }

  > .visualization {
    background-color: $visualizations !important;
  }

  .title {
    border-radius : inherit;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    text-align: center;
    padding: 5px;
    border: 1px solid var(--almostWhite);
    background-color: var(--dark);
    color: var(--white);
  }
  
  div {
    p {
      margin-top: 4px;
      margin-bottom: 4px;
    }
  }
}
