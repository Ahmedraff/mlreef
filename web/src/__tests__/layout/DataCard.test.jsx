import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import DataCard from 'components/layout/DataCard';

const title = 'Data';
const content = [
  { text: 'Files selected from path' },
  {
    text: 'Folder Name',
    isLink: true,
    href: '/some-path',
  },
  { text: 'from' },
  { text: 'Branch Name' },
];

const setup = () => {
  const testWrapper = shallow(
    <DataCard title={title} linesOfContent={content} styleClasses="model" />,
  );
  return testWrapper;
};

test('assert that snapshot matches', () => {
  const snapShot = renderer.create(
    <DataCard title={title} linesOfContent={content} />,
  ).toJSON();
  expect(snapShot).toMatchSnapshot();
});

describe('Data instance details contains basic UI elements', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('assert the data card title', () => {
    expect(wrapper.find('.title').text()).toEqual('Data');
  });

  it('assert that lines of content are rendered correctly', () => {
    expect(wrapper.find('.line')).toHaveLength(3);
    expect(wrapper.find('a')).toHaveLength(1);
  });
});
