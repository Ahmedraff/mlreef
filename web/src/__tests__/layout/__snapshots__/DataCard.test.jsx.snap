// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assert that snapshot matches 1`] = `
<div
  className="data-card ml-2 mr-2"
>
  <div
    className="title "
  >
    <p>
      <b>
        Data
      </b>
    </p>
  </div>
  <div
    className="data-card-rows"
  >
    <p
      className="line"
    >
      Files selected from path
    </p>
    <div
      className="link"
    >
      <a
        href="/some-path"
        rel="noopener noreferrer"
        target="_blank"
      >
        <b>
          Folder Name
        </b>
      </a>
    </div>
    <p
      className="line"
    >
      from
    </p>
    <p
      className="line"
    >
      Branch Name
    </p>
  </div>
</div>
`;
